gulpfile.js

```diff
@@ -0,0 +1,88 @@
+const autoprefixer = require('gulp-autoprefixer');
+const bs = require('browser-sync').create('amagaki-browsersync');
+const esbuild = require('esbuild');
+const gulp = require('gulp');
+const sass = require('gulp-sass');
+const {exec} = require('child_process');
+sass.compiler = require('sass');
+
+const ENTRIES = {
+  js: {
+    src: ['./src/ts/main.js'],
+    out: './dist/js/main.min.js',
+    watch: ['./src/ts/**/*.ts'],
+  },
+  sass: {
+    src: ['./src/sass/**/*.sass'],
+    out: './dist/css/',
+    watch: ['./src/sass/**/*.sass'],
+  },
+};
+
+const runEsBuild = async prod => {
+  return new Promise(resolve => {
+    exec('tsc', async (error, stderr) => {
```

Comment:
i still think it'd be preferable in dev to not have tsc run in series with esbuild, and instead have them run in parallel, because the ts->js transpilation is happening twice here.

so like i mentioned in another comment, i think it'd either be better to have the two run in parallel or have the esbuild rule use files from the tsc generated output.

(you could get fancy and have a gulp task that just runs tsc with the noEmit flag, and then use `gulp.series()` in prod and `gulp.parallel()` in dev. but i know that complicates this gulpfile, so up to you.)

Reply:
>> have the esbuild rule use files from the tsc generated output.

`entryPoints: ENTRIES.js.src,` is the outputted JS file from the tsc build so I believe the esBuild is building from the tsc output (the js output from tsc).

One reason I wanted esBuild to run in series even on dev is that, esBuild will build even if there are type errors and running this in parallel makes it easy to miss these such errors.

Reply:
k sounds good. can you add some comment to the ENTRIES.js.src line that says something about the entry file being the generated output from tsc? (otherwise a new dev looking at it may think it's a typo.) or instead of having the tsc-compiled files output to the `src/` folder, maybe have it output to `build/` so it's more clear.

Reply:
Great!  I renamed `ENTRIES.js.src` to `ENTRIES.js.tsc_ouput` to make it clearer that it is references the JS file of your tsc build (set in your tsconfig).

I also added some comment on why we are building via tsc first.